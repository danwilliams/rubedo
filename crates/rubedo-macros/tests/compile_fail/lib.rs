#![allow(non_snake_case)]
#![allow(unused_imports)]

//		Tests																											

fn main() {
	use rubedo_macros::*;
	use std::net::IpAddr;
	
	//		ip!																	
	//	These tests are here as it is is not possible to test a proc_macro in
	//	the same crate as it is defined.
	fn ip__str_invalid_1() {
		ip!("1.2.3");
	}
	fn ip__str_invalid_2() {
		ip!("1.2.3.4.5");
	}
	fn ip__str_invalid_3() {
		ip!("1.2.3.999");
	}
	fn ip__args_invalid_1() {
		ip!(1, 2, 3);
	}
	fn ip__args_invalid_2() {
		ip!(1, 2, 3, 4, 5);
	}
	fn ip__args_invalid_3() {
		ip!(1, 2, 3, 999);
	}
	fn ip__raw_invalid_1() {
		ip!(1.2.3);
	}
	fn ip__raw_invalid_2() {
		ip!(1.2.3.4.5);
	}
	fn ip__raw_invalid_3() {
		ip!(1.2.3.999);
	}
	fn ip__mixed_invalid_1() {
		ip!(1.2,3.4);
	}
	fn ip__mixed_invalid_2() {
		ip!(1,2.3.4);
	}
}


