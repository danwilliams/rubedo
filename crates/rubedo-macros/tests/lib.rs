#![allow(unused_crate_dependencies)]

//	Lints specifically disabled for integration tests
#![cfg_attr(test, allow(
	non_snake_case,
	clippy::cast_lossless,
	clippy::cast_precision_loss,
	clippy::cognitive_complexity,
	clippy::default_numeric_fallback,
	clippy::exhaustive_enums,
	clippy::exhaustive_structs,
	clippy::expect_used,
	clippy::indexing_slicing,
	clippy::let_underscore_must_use,
	clippy::let_underscore_untyped,
	clippy::missing_assert_message,
	clippy::missing_panics_doc,
	clippy::must_use_candidate,
	clippy::panic,
	clippy::print_stdout,
	clippy::tests_outside_test_module,
	clippy::unwrap_in_result,
	clippy::unwrap_used,
))]



//		Tests

//		ip!																		
#[cfg(test)]
mod ip {
	use rubedo_macros::*;
	use core::net::Ipv4Addr;
	
	//		ip!																	
	#[test]
	fn ip__empty_1() {
		assert_eq!(ip!(), Ipv4Addr::new(0, 0, 0, 0));
	}
	#[test]
	fn ip__empty_2() {
		assert_eq!(ip!(""), Ipv4Addr::new(0, 0, 0, 0));
	}
	#[test]
	fn ip__str_valid() {
		assert_eq!(ip!("1.2.3.4"), Ipv4Addr::new(1, 2, 3, 4));
	}
	#[test]
	fn ip__args_valid() {
		assert_eq!(ip!(1, 2, 3, 4), Ipv4Addr::new(1, 2, 3, 4));
	}
	#[test]
	fn ip__raw_valid_1() {
		assert_eq!(ip!(1.2.3.4), Ipv4Addr::new(1, 2, 3, 4));
	}
	#[test]
	fn ip__raw_valid_2() {
		assert_eq!(ip!(1 . 2 . 3 . 4), Ipv4Addr::new(1, 2, 3, 4));
	}
	#[test]
	fn ip__mixed_invalid() {
		//	We don't really want this, but it's a side-effect of the parser.
		assert_eq!(ip!("1.2".3.4), Ipv4Addr::new(1, 2, 3, 4));
	}
	#[test]
	fn compile_fail() {
		trybuild::TestCases::new().compile_fail("tests/compile_fail/lib.rs");
	}
}


