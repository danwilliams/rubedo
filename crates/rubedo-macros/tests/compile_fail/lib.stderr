error: proc macro panicked
  --> tests/compile_fail/lib.rs:14:3
   |
14 |         ip!("1.2.3");
   |         ^^^^^^^^^^^^
   |
   = help: message: Invalid IP address

error: proc macro panicked
  --> tests/compile_fail/lib.rs:17:3
   |
17 |         ip!("1.2.3.4.5");
   |         ^^^^^^^^^^^^^^^^
   |
   = help: message: Invalid IP address

error: proc macro panicked
  --> tests/compile_fail/lib.rs:20:3
   |
20 |         ip!("1.2.3.999");
   |         ^^^^^^^^^^^^^^^^
   |
   = help: message: Invalid IP address: AddrParseError(Ip)

error: proc macro panicked
  --> tests/compile_fail/lib.rs:23:3
   |
23 |         ip!(1, 2, 3);
   |         ^^^^^^^^^^^^
   |
   = help: message: Invalid IP address

error: proc macro panicked
  --> tests/compile_fail/lib.rs:26:3
   |
26 |         ip!(1, 2, 3, 4, 5);
   |         ^^^^^^^^^^^^^^^^^^
   |
   = help: message: Invalid IP address

error: proc macro panicked
  --> tests/compile_fail/lib.rs:29:3
   |
29 |         ip!(1, 2, 3, 999);
   |         ^^^^^^^^^^^^^^^^^
   |
   = help: message: Invalid IP address: AddrParseError(Ip)

error: proc macro panicked
  --> tests/compile_fail/lib.rs:32:3
   |
32 |         ip!(1.2.3);
   |         ^^^^^^^^^^
   |
   = help: message: Invalid IP address

error: proc macro panicked
  --> tests/compile_fail/lib.rs:35:3
   |
35 |         ip!(1.2.3.4.5);
   |         ^^^^^^^^^^^^^^
   |
   = help: message: Invalid IP address

error: proc macro panicked
  --> tests/compile_fail/lib.rs:38:3
   |
38 |         ip!(1.2.3.999);
   |         ^^^^^^^^^^^^^^
   |
   = help: message: Invalid IP address: AddrParseError(Ip)

error: proc macro panicked
  --> tests/compile_fail/lib.rs:41:3
   |
41 |         ip!(1.2,3.4);
   |         ^^^^^^^^^^^^
   |
   = help: message: Invalid IP address

error: proc macro panicked
  --> tests/compile_fail/lib.rs:44:3
   |
44 |         ip!(1,2.3.4);
   |         ^^^^^^^^^^^^
   |
   = help: message: Invalid IP address
